File: backend/backend/Areas/Main/Models/TaskNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Areas.Main.Models;

public class TaskNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public int TaskId { get; set; }
    public Tasks Task { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/task_notes.dart
import 'package:frontend/models/tasks.dart';
import 'package:frontend/models/note.dart';

class TaskNotes {
  int? id;
  Tasks? task;
  Note? note;
  int? taskId;
  int? noteId;

  TaskNotes({
    this.id,
    this.task,
    this.note,
    this.taskId,
    this.noteId
  });

  factory TaskNotes.fromJson(Map<String, dynamic> json) {
    return TaskNotes(
      id: json['id'],
      taskId: json['taskId'],
      task: json['task'] != null ? Tasks.fromJson(json['task']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'task': task,
      'note': note,
      'noteId': noteId,
      'taskId': taskId
    };
  }
}